EpikCord/client/http_client.py

```diff
@@ -1,312 +1,233 @@
+    def clear(self):
+        ...
 
-    def __hash__(self):
-        return id(self)
+    def set(self):
+        ...
 
+    def __eq__(self, other: MockBucket):
+        if isinstance(other, MockBucket):
+            return True
+        return False
 
-class DiscordWSMessage:
-    def __init__(self, *, data, type, extra):
-        self.data = data
-        self.type = type
-        self.extra = extra
 
-    def json(self) -> Any:
-        return json.loads(self.data)
-
-
-class GatewayWebsocket(ClientWebSocketResponse):
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-        self.buffer: bytearray = bytearray()
-        self.inflator = zlib.decompressobj()
+class Bucket:
+    def __init__(self, *, bucket_hash: str):
+        """
+        Parameters
+        ----------
+        bucket_hash: str
+            The bucket hash of the bucket provided by Discord.
+
+        Attributes
+        ----------
+        bucket_hash: str
+            The bucket hash of the bucket provided by Discord.
+        event: asyncio.Event
+            The event that is used to wait for the bucket to be set.
+        """
+        self.hash: str = bucket_hash
+        self.event: asyncio.Event = asyncio.Event()
+        self.event.set()
 
-    async def receive(self, *args, **kwargs):
-        ws_message = await super().receive(*args, **kwargs)
-        message = ws_message.data
+    async def wait(self):
+        """
+        Waits for the bucket to be set.
+        """
+        await self.event.wait()
 
-        if isinstance(message, bytes):
+    def set(self):
+        """
+        Sets the bucket.
+        """
+        self.event.set()
 
-            self.buffer.extend(message)
+    def clear(self):
+        """
+        Clears the bucket.
+        """
+        self.event.clear()
 
-            if len(message) < 4 or message[-4:] != b"\x00\x00\xff\xff":
-                return
+    def __eq__(self, other: Bucket):
+        if isinstance(other, Bucket):
+            return self.hash == other.hash
+        return False
 
-            message = self.inflator.decompress(self.buffer)
 
-            message = message.decode("utf-8")
-            self.buffer: bytearray = bytearray()
+class TopLevelBucket(Bucket):
+    def __init__(self, *, major_parameters: Dict[str, Any]):
+        """
+        Parameters
+        ----------
+        internal_hash: str
+            The internal hash of the bucket.
+        external_hash: str
+            The external hash of the bucket.
+        major_parameters: Dict[str, Any]
+            The major parameters for this Bucket
+
+        Attributes
+        ----------
+        internal_hash: str
+            The internal hash of the bucket.
+        external_hash: str
+            The external hash of the bucket.
+        major_parameters: Dict[str, Any]
+            The major parameters for this Bucket
+        """
+        super().__init__(internal_hash=internal_hash, external_hash=external_hash)
```

Comment:
These variables aren't defined?

Reply:
Yes this was old code that somehow made it past, I'll change it soon.